torch/_inductor/fx_passes/reinplace.py

```diff
@@ -460,20 +462,36 @@ def is_meta_only_user(node):
                 return True
         return False
 
+    def _overlap(ls) -> bool:
+        try:
+            return len(compute_overlapping_tensors(ls)) != 0
+        except GuardOnDataDependentSymNode:
+            # If we fail with data dependent error we assume they all overlap.
```

Comment:
is `ls` guaranteed to be a list of tensors that all share the same `storage` object?

(if not, our fallback here can just be to return true if any of the tensors share the same storage, and false otherwise)

Reply:
for the current usage it will only receive things that have the same storage , mhmm yeh maybe why i got some faluires when i used it on the other location. good point i can add a comment or just harden it for future usages

Reply:
add comment